pkg/generated/openapi/def.bzl

```diff
@@ -14,17 +15,17 @@ def openapi_library(name, tags, srcs, openapi_targets=[], vendor_targets=[]):
   )
   go_genrule(
       name = "zz_generated.openapi",
-      srcs = srcs + ["//hack/boilerplate:boilerplate.go.txt"],
+      srcs = srcs + ["//" + openapi_pkg_prefix + "hack/boilerplate:boilerplate.go.txt"],
       outs = ["zz_generated.openapi.go"],
       cmd = " ".join([
         "$(location //vendor/k8s.io/code-generator/cmd/openapi-gen)",
         "--v 1",
         "--logtostderr",
-        "--go-header-file $(location //hack/boilerplate:boilerplate.go.txt)",
+        "--go-header-file $(location //" + openapi_pkg_prefix + "hack/boilerplate:boilerplate.go.txt)",
         "--output-file-base zz_generated.openapi",
-        "--output-package k8s.io/kubernetes/pkg/generated/openapi",
-        "--input-dirs " + ",".join(["k8s.io/kubernetes/" + target for target in openapi_targets] + ["k8s.io/kubernetes/vendor/" + target for target in vendor_targets]),
-        "&& cp pkg/generated/openapi/zz_generated.openapi.go $(location :zz_generated.openapi.go)",
+        "--output-package " + openapi_output_prefix + "k8s.io/kubernetes/pkg/generated/openapi",
+        "--input-dirs " + ",".join([openapi_target_prefix + target for target in openapi_targets] + ["k8s.io/kubernetes/vendor/" + target for target in vendor_targets]),
+        "&& cp " + openapi_pkg_prefix + "pkg/generated/openapi/zz_generated.openapi.go $(location :zz_generated.openapi.go)",
       ]),
       go_deps = deps,
       tools = ["//vendor/k8s.io/code-generator/cmd/openapi-gen"],
```

Comment:
This could cause problems for generalizability as well.

Reply:
I have a related PR that allows ``openapi-gen`` to generate into a vendored path: https://github.com/kubernetes/kube-openapi/pull/14